Mathlib/RingTheory/AdjoinRoot.lean

```diff
@@ -105,11 +105,13 @@ def of : R →+* AdjoinRoot f :=
   (mk f).comp C
 #align adjoin_root.of AdjoinRoot.of
 
-instance instSMulAdjoinRoot [DistribSMul S R] [IsScalarTower S R R] : SMul S (AdjoinRoot f) :=
+instance smul [DistribSMul S R] [IsScalarTower S R R] : SMul S (AdjoinRoot f) :=
   Submodule.Quotient.instSMul' _
 
 instance [DistribSMul S R] [IsScalarTower S R R] : DistribSMul S (AdjoinRoot f) :=
-  Submodule.Quotient.distribSMul' _
+  --TODO: add reference to library note in PR #7432
+  { Submodule.Quotient.distribSMul' _ with
```

Comment:
`distribSMul'` uses `Function.Surjective` which is probably the underlying problem

Reply:
I don't think so. For example, the change in #6803 to the group instance made a difference even though the underlying instance on `Con` was defined nicely. Maybe defining them nicely all the way back also makes a difference.